app/assets/javascripts/create_idea.js

```diff
@@ -7,22 +7,12 @@ $(document).ready(function() {
       body: $("#idea-body").val()
     }}
 
-    $.ajax({
-      url: '/api/v1/ideas',
-      data: ideaParams,
-      type: 'post',
-      success: function(idea){
-        var formattedIdea = "<tr data-id=" + idea.id
-        + " id='idea'>" + "<td contenteditable='true' class='title-text'>"
-        + idea.title + "</td>" + "<td contenteditable='true' class='body-text'>"
-        + idea.body + "</td>" + "<td class='quality-text'>"
-        + idea.quality
-        + "</td>"
-        + "<td><input type='button' class='upvote-idea' value='Thumbs Up'><input type='button' class='downvote-idea' value='Thumbs Down'></td><td><input type='button' value='Delete' class='destroy-idea'></td></tr>";
-        $(".all-ideas").prepend(formattedIdea)
-        $(".idea-title").val("")
-        $(".idea-body").val("")
-      }
-    })
-  })
-})
+    $.post('/api/v1/ideas', ideaParams)
+    .then(createIdeaHTML)
+    .then(renderIdea)
+    .fail(handleError)
+
+    e.preventDefault();
```

Comment:
I would move the preventDefault to the top of the event. Not sure if that's my style preference and now just habit, or if it's a good practice because of its functionality.